src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -575,25 +575,9 @@ internal sealed override CustomAttributesBag<CSharpAttributeData> GetAttributesB
         /// <summary>
         /// Binds attributes applied to this parameter.
         /// </summary>
-        public ImmutableArray<BoundAttribute> BindParameterAttributes()
+        public ImmutableArray<(CSharpAttributeData, BoundAttribute)> BindParameterAttributes()
         {
-            var binder = new ContextualAttributeBinder(WithTypeParametersBinderOpt, this);
-            var boundAttributeArrayBuilder = ArrayBuilder<BoundAttribute>.GetInstance();
-            foreach (var attributeListSyntaxList in GetAttributeDeclarations())
-            {
-                foreach (var attributeListSyntax in attributeListSyntaxList)
-                {
-                    foreach (var attributeSyntax in attributeListSyntax.Attributes)
-                    {
-                        var boundType = binder.BindType(attributeSyntax.Name, BindingDiagnosticBag.Discarded);
-                        var boundTypeSymbol = (NamedTypeSymbol)boundType.Type;
-                        var boundAttribute = new ExecutableCodeBinder(attributeSyntax, binder.ContainingMemberOrLambda, binder)
-                            .BindAttribute(attributeSyntax, boundTypeSymbol, this, BindingDiagnosticBag.Discarded);
-                        boundAttributeArrayBuilder.Add(boundAttribute);
-                    }
-                }
-            }
-            return boundAttributeArrayBuilder.ToImmutableAndFree();
+            return BindAttributes(GetAttributeDeclarations(), WithTypeParametersBinderOpt);
```

Comment:
>WithTypeParametersBinderOpt [](http://example.com/codeflow?start=62&length=27)

I'm wondering about old code. What's a scenario where `WithTypeParametersBinderOpt` would be null?
If this is reachable, then the parameter should be marked as nullable in `BindAttributes`.

Reply:
Not sure when it can be `null`, but I can do the same thing as `GetAttributesBag` does when binding attributes: 7c19b43548cec1368c4eb803e92b0fd94e56fa78

Reply:
OK, seems that `SourceComplexParameterSymbolBase.WithTypeParameterOpt == null` happens for parameters of `SourcePropertySymbol`. But `SourceMethodSymbolWithAttributes.OuterBinder == null` never happens as all current implementations provide non-`null` binder (although the abstract property type is nullable).

For the former case, I added an explicit test of the `BindParameterAttributes()` method to verify it behaves correctly when `WithTypeParameterOpt == null`: 1892e8cca890f5f4d8299578cb1518619758a3a6. This is not a normal integration test as I don't think this can be hit directly though definite assignment analysis. Also not sure this is the correct place for the test.